src/sentry/monitoring/queues.py

```diff
@@ -94,3 +116,66 @@ def get_queue_by_name(name):
     backend = get_backend_for_broker(settings.BROKER_URL)
 except KeyError:
     backend = None
+
+
+queue_monitoring_cluster = redis.redis_clusters.get(CLUSTER_NAME)
+
+
+def _unhealthy_queue_key(queue_name: str) -> str:
+    return f"{KEY_NAME}:{queue_name}"
+
+
+def is_queue_healthy(queue_name: str) -> bool:
+    if not options.get("backpressure.enable_monitor_queues"):
+        return True
+    # check if queue is healthy by pinging Redis
+    return not queue_monitoring_cluster.exists(_unhealthy_queue_key(queue_name))
+
+
+def _is_healthy(queue_size) -> bool:
+    return queue_size < UNHEALTHY_QUEUE_SIZE_THRESHOLD
+
+
+def _update_queue_stats(redis_cluster, unhealthy) -> None:
+    if not unhealthy:
+        return
+
+    with sentry_sdk.push_scope() as scope:
+        scope.set_extra("unhealthy_queues", unhealthy)
+        sentry_sdk.capture_message("Queues are unhealthy")
+
+    with redis_cluster.pipeline(transaction=True) as pipeline:
+        # can't use mset because it doesn't support expiry
+        for queue in unhealthy:
+            pipeline.set(_unhealthy_queue_key(queue), "1", ex=UNHEALTHY_QUEUE_CHECK_INTERVAL)
+        pipeline.execute()
+
+
+def _run_queue_stats_updater(redis_cluster: str) -> None:
+    # bonus point if we manage to use asyncio and launch all tasks at once
+    # in case we have many queues to check
+    cluster = redis.redis_clusters.get(redis_cluster)
+
+    queue_history = {queue: 0 for queue in QUEUES}
+    while True:
+        if not options.get("backpressure.enable_monitor_queues"):
+            continue
+        new_sizes = backend.bulk_get_sizes(QUEUES)
+        for (queue, size) in new_sizes:
+            if _is_healthy(size):
+                queue_history[queue] = 0
+            else:
+                queue_history[queue] += 1
+
+        unhealthy = [
+            queue for (queue, count) in queue_history if count >= UNHEALTHY_QUEUE_STRIKE_THRESHOLD
```

Comment:
> This means, you could instead sum up all queue counts and compare those with a global threshold. That should be a bit closer to what we're gonna measure with memory later.

That means multiple queues are running on a single rabbit server, I guess thats a general assumption, because why not :-D